jgrapht-core/src/main/java/org/jgrapht/alg/IndependentSet.java

```diff
@@ -0,0 +1,73 @@
+package org.jgrapht.alg;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.alg.interfaces.MinimumVertexCoverAlgorithm.VertexCover;
+import org.jgrapht.alg.interfaces.MinimumWeightedVertexCoverAlgorithm;
+import org.jgrapht.alg.vertexcover.RecursiveExactVCImpl;
+
+
+/**
+ * Finds a maximum (weight) independent set in a undirected graph. 
+ * 
+ * The implementation relies on the fact that if V' is a minimum 
+ * weight vertex cover for graph G = (V,E) with weight function
+ * w: V -> |R_+, then V-V' is a maximum weight independent set in G. 
+ *
+ * @author Nils Olberg
+ */
+
+public abstract class IndependentSet {
```

Comment:
This is just one of the possible ways to find independent sets, so we may end up in a similar situation (needing to split interface from implementation).